lib/google_scraper_web/live/homepage_live/index.ex

```diff
@@ -56,21 +61,59 @@ defmodule GoogleScraperWeb.HomepageLive.Index do
 
   @impl Phoenix.LiveView
   def handle_info({:fetch_from_google, contents}, socket) do
-    if Enum.count(contents) < 100 do
+    if Enum.count(contents) < @keyword_limit do
       user_id = socket.assigns.user.id
+      view = self()
 
-      contents
-      |> GoogleScraper.fetch_results(user_id)
-      |> Keywords.maybe_insert_keywords()
+      Task.start(fn ->
+        GoogleScraper.fetch_results(contents, user_id)
+        send(view, :fetch_from_google_complete)
+      end)
 
-      {:noreply,
-       assign(socket, loading: false, keywords: Keywords.list_keywords_by_user(user_id))}
+      {:noreply, socket}
     else
-      IO.puts("LARGER THAN 100")
-      {:noreply, assign(socket, loading: false)}
+      socket =
+        socket
+        |> assign(loading: false, error_message: "Your CSV file has more than 100 keywords")
+
+      {:noreply, socket}
     end
   end
 
+  def handle_info(:fetch_from_google_complete, socket) do
+    {:noreply, assign(socket, loading: false)}
+  end
+
+  @impl Phoenix.LiveView
+  def handle_info(
+        {:keyword_created,
+         %{
+           name: name,
+           total_advertisers: total_advertisers,
+           total_links: total_links,
+           total_results: total_results,
+           user_id: user_id
+         }},
```

Comment:
Not sure if is necessary to expand the whole keyword map since `:keyword_created` is unique enough (so no need for patten matching) and the whole map can be used in the update:

```elixir
def handle_info({:keyword_created, keyword, socket) do
  {:noreply,
     update(socket, :keywords, fn keywords ->
       keywords ++ keyword
     end)}
end
```

It might be a code style preference at this level 🙈

Reply:
> the whole map can be used in the update

Maybe I'm super wrong on this, I need to investigate more. I don't like speaking about things I don't know. But I don't know if this map is being **copied** or **referenced** from one place to another. If it's being copied then I should say this map is very big. I think it might has a size of 1 MB in memory. I don't want to overload the memory copying (duplicating) unnecessary data, like the `:html_content` value (the big one).

The reason I pattern matched is because I wanted to have a smaller/narrowed map with a smaller size.

There's a high chance I'm speaking nonsense 🥲.

For example, suppose this:

```Elixir
iex(1)> big_map = %{a: "a", b: "b", c: "c"}
%{a: "a", b: "b", c: "c"}

iex(2)> %{a: value} = big_map
%{a: "a", b: "b", c: "c"}

iex(3)> small_map = %{a: value} 
%{a: "a"}
```

Definitely `small_map` will have a smaller size.

**That was my scheme of thought for doing this**. It was a trade-off. You don't overload memory but you need to sacrifice some code readability